test/e2e/flowaggregator_test.go

```diff
@@ -102,21 +102,45 @@ const (
 )
 
 func TestFlowAggregator(t *testing.T) {
-	skipIfDualStackCluster(t)
-	data, isIPv6, err := setupTestWithIPFIXCollector(t)
+	data, v4Enabled, v6Enabled, err := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
-	podAIP, podBIP, podCIP, svcB, svcC, err := createPerftestPods(data, isIPv6)
+	podAIP, podBIP, podCIP, err := createPerftestPods(data)
 	if err != nil {
-		t.Fatalf("Error when creating perftest pods and services: %v", err)
+		t.Fatalf("Error when creating perftest pods: %v", err)
 	}
-	// Wait for the Service to be realized.
-	time.Sleep(3 * time.Second)
 
+	if v4Enabled {
+		svcB, svcC, err := createPerftestServices(data, false)
+		if err != nil {
+			t.Fatalf("Error when creating IPv4 perftest services: %v", err)
+		}
+		// Wait for the Service to be realized.
+		time.Sleep(3 * time.Second)
+
+		runTests(t, data, podAIP, podBIP, podCIP, svcB, svcC, false)
+		deletePerftestServices(data)
+	}
+
+	if v6Enabled {
+		svcB, svcC, err := createPerftestServices(data, true)
+		if err != nil {
+			t.Fatalf("Error when creating IPv6 perftest services: %v", err)
+		}
+		// Wait for the Service to be realized.
+		time.Sleep(3 * time.Second)
+
+		runTests(t, data, podAIP, podBIP, podCIP, svcB, svcC, true)
+		deletePerftestServices(data)
+	}
+}
+
+func runTests(t *testing.T, data *TestData, podAIP *PodIPs, podBIP *PodIPs, podCIP *PodIPs, svcB *corev1.Service, svcC *corev1.Service, isIPv6 bool) {
```

Comment:
why is this helper method not in charge of creating the test resources (pods, services)? Why the need to pass them as parameters?

Reply:
Hi Antonin, after changing to use subtests, only pod IPs are passed as parameters to the testHelper method. The reason is that we would like to reuse perf pods in both IPv4 and IPv6 tests, while perf services need to be created for IPv4 and IPv6 differently.